reporting/api/src/main/java/org/jboss/windup/reporting/service/ClassificationServiceCache.java

```diff
@@ -0,0 +1,71 @@
+package org.jboss.windup.reporting.service;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.commons.collections.map.LRUMap;
+import org.jboss.windup.config.AbstractRuleLifecycleListener;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.RuleLifecycleListener;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.reporting.model.ClassificationModel;
+
+import com.tinkerpop.blueprints.Vertex;
+import com.tinkerpop.gremlin.java.GremlinPipeline;
+
+/**
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class ClassificationServiceCache extends AbstractRuleLifecycleListener implements RuleLifecycleListener
```

Comment:
This looks good. It's nice that you found out that the for cycle was causing the issue. Is there a specific reason why you use a) classif_id_file_id => boolean     instead of     b) classif_id => Set(file_id) ?